### DISCUSSION TITLE
Multi-select component

### DISCUSSION THREAD

**Discussion by scottanthonymurray** [indiana-university/rivet-source#409] [Feature and enhancement requests]
We've seen quite a few requests for a multi-select component in the vein of the popular jQuery plugin [Select2](https://select2.org/), or this similar vanilla JS version called [Tagify](https://yaireo.github.io/tagify/).

The multi-select UI is a pretty popular UI solution for filter/searching through a bunch of options/records/filters, selecting one result and repeating until you have a list of selections. You see this a lot with interfaces that allow users to create, or select from a list of, "Tags". It's a much more elegant alternative to the native browser `<select>` element with a `multiple` attribute applied.

It's probably a good idea to explore a third-party option for this that we could theme and offer as a Rivet Add-on, but we'll need to spend some time up front reviewing for accessibility concerns.

I would also be interested in see some actual use cases where folks are wanting to use this kind of UI in their apps to get a good idea of what types of features need to be supported. Feel free to post screenshots, etc. here if you have them.

**Comment by afraze9** [comment]
+1

I would use this in an 'Admin' interface to filter data grids.  

Tagging with Multi-value select boxes would be especially nice
https://select2.org/tagging#tagging-with-multi-value-select-boxes

Alternatively, you could extend on your Segmented Buttons such that it resembles the idea attached in the screenshot to this post.  (Not exactly like it but it should give you a general idea.  Move the dropdown left and in addition to the checkmark possibly add a background color to add contrast)

![screenshot-localhost-44316-2019 02 06-15-12-59](https://user-images.githubusercontent.com/39129047/52371385-5a606700-2a23-11e9-8b2f-7fcc1246bb3c.png)

**Comment by afraze9** [comment]
Here is a screenshot of one I've created extending Rivet...

![screenshot-localhost-44316-2019 02 11-15-02-38](https://user-images.githubusercontent.com/39129047/52590012-485d3a80-2e0e-11e9-96dc-11a09fb45b66.png)
